### PR TITLE
Add tests of average for nullable column

### PR DESCRIPTION
**Pull request by zaki50**
Added tests for aggregation methods in `RealmQuery` for partial null rows .

`RealmQueryTest.testSumGivesCorrectValueForPartialNullRows()` and `RealmQueryTest.testAvgGivesCorrectValueForAllNonNullRows()` tests wrong value for now.

These tests must be updated after core's fix.

related to #1803

### CODE REVIEW

**Comment by cmelchior** [comment]
Instead of letting the unit test fail, you could also let it detect the wrong value. That way we could merge these unit tests right away instead of letting them wait?

**Comment by zaki50** [comment]
@cmelchior sure

**Comment by zaki50** [comment]
please review this @realm/java

File: realm/realm-library/src/androidTest/java/io/realm/RealmQueryTest.java [hunk 1]
```diff
@@ -1011,6 +1011,27 @@ public void testMinValueForAllNullColumns() {
         assertNull(query.minimumDate(NullTypes.FIELD_DATE_NULL));
     }
 
+    // Test min on columns with partial null rows
+    public void testMinGivesCorrectValueForAllNonNullRows() {
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/RealmQueryTest.java]: I would remove "GivesCorrect" from these. Hopefully that should be implicit :) So `testMinForAllNonNullRows`, same for the others.

File: realm/realm-library/src/androidTest/java/io/realm/RealmQueryTest.java [hunk 2]
```diff
@@ -1049,6 +1092,28 @@ public void testAverageValueForAllNullColumns() {
         assertEquals(0d, query.average(NullTypes.FIELD_DOUBLE_NULL), 0d);
     }
 
+    // Test average on columns with partial null rows
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/RealmQueryTest.java]: This description doesn't seem to match the unit test (seems like copy paste from below?)

**Comment by cmelchior** [comment]
Apart from a few minor naming/doc issues it looks good :+1:

File: realm/realm-library/src/androidTest/java/io/realm/RealmQueryTest.java (hunk 2 above)
Comment (zaki50) [inline realm/realm-library/src/androidTest/java/io/realm/RealmQueryTest.java]: done

File: realm/realm-library/src/androidTest/java/io/realm/RealmQueryTest.java (hunk 1 above)
Comment (zaki50) [inline realm/realm-library/src/androidTest/java/io/realm/RealmQueryTest.java]: done

**Comment by kneth** [comment]
:+1:

**Pull request merged**